ui/tests/integration/components/select-test.js

```diff
@@ -0,0 +1,77 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render, fillIn } from '@ember/test-helpers';
+import hbs from 'htmlbars-inline-precompile';
+import sinon from 'sinon';
+
+const OPTIONS = ['foo', 'bar', 'baz'];
+const LABEL = 'Boop';
+
+module('Integration | Component | Select', function(hooks) {
+  setupRenderingTest(hooks);
+
+  hooks.beforeEach(function() {
+    this.set('options', OPTIONS);
+    this.set('label', LABEL);
+    this.set('name', 'foo');
+  });
+
+  test('it renders', async function(assert) {
+    await render(hbs`<Select @options={{options}} @label={{label}} @name={{name}}/>`);
+    assert.dom('[data-test-select-label]').hasText('Boop');
+    assert.dom('[data-test-select="foo"]').exists();
+  });
+
+  test('it renders when options is an array of strings', async function(assert) {
+    await render(hbs`<Select @options={{options}} @label={{label}} @name={{name}}/>`);
+
+    assert.dom('[data-test-select="foo"]').hasValue('foo');
+    assert.equal(this.element.querySelector('[data-test-select="foo"]').options.length, 3);
+  });
+
+  test('it renders when options is an array of objects', async function(assert) {
+    const objectOptions = [{ value: 'berry', label: 'Berry' }, { value: 'cherry', label: 'Cherry' }];
+    this.set('options', objectOptions);
+    await render(hbs`<Select @options={{options}} @label={{label}} @name={{name}}/>`);
+
+    assert.dom('[data-test-select="foo"]').hasValue('berry');
+    assert.equal(this.element.querySelector('[data-test-select="foo"]').options.length, 2);
+  });
+
+  test('it renders when options is an array of custom objects', async function(assert) {
+    const objectOptions = [{ day: 'mon', fullDay: 'Monday' }, { day: 'tues', fullDay: 'Tuesday' }];
+    this.setProperties({
+      options: objectOptions,
+      valueAttribute: 'day',
+      labelAttribute: 'fullDay',
+    });
+
+    await render(
+      hbs`
+        <Select
+          @options={{options}}
+          @label={{label}}
+          @name={{name}}
+          @valueAttribute={{valueAttribute}}
+          @labelAttribute={{labelAttribute}}/>`
+    );
+
+    assert.dom('[data-test-select="foo"]').hasValue('mon');
+    assert.equal(this.element.querySelector('[data-test-select="foo"]').options[1].textContent, 'Tuesday');
+  });
+
+  test('it renders the selectedItem as selected by default', async function(assert) {
```

Comment:
Can we add a test for passing the selectedItem as an object as well?

Reply:
yes!

Reply:
`selectedItem` should only be a string since the `onchange` only passes back the event `target.value`. i'll update the docs to specify that!